src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeBlockTagsGroupCheck.java

```diff
@@ -0,0 +1,243 @@
+ *     return false;
+ * }
+ * </pre>
+ * <p>
+ *  Valid javadoc should have a blank line separating the parameters, return, throw, or
+ *  other tags like in the example below.
+ *  </p>
+ *  <pre>
+ *  &#47;**
+ *  * testMethod's javadoc.
+ *  *
+ *  * &#64;param firstParam
+ *  * &#64;return something
+ *  *&#47;
+ *  public boolean testMethod(int firstParam) {
+ *      return false;
+ *  }
+ *  </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.line.before}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.36
+ */
+@StatelessCheck
+public class RequireEmptyLineBeforeBlockTagsGroupCheck extends AbstractJavadocCheck {
+
+    /**
+     * The key in "messages.properties" for the message that describes a tag in javadoc
+     * requiring an empty line before it.
+     */
+    public static final String MSG_JAVADOC_TAG_LINE_BEFORE = "javadoc.tag.line.before";
+
+    /**
+     * Case when space separates the tag and the asterisk like in the below example.
+     * <pre>
+     *  /**
+     *   * &#64;param noSpace there is no space here
+     * </pre>
+     */
+    private static final List<Integer> ONLY_TAG_VARIATION_1 = Arrays.asList(
+            JavadocTokenTypes.WS,
+            JavadocTokenTypes.LEADING_ASTERISK,
+            JavadocTokenTypes.NEWLINE);
+
+    /**
+     * Case when no space separates the tag and the asterisk like in the below example.
+     * <pre>
+     *  /**
+     *   *&#64;param noSpace there is no space here
+     * </pre>
+     */
+    private static final List<Integer> ONLY_TAG_VARIATION_2 = Arrays.asList(
+            JavadocTokenTypes.LEADING_ASTERISK,
+            JavadocTokenTypes.NEWLINE);
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC_TAG,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    /**
+     * Logs when there is no empty line before the tag.
+     *
+     * @param tagNode the at tag node to check for an empty space before it.
+     */
+    @Override
+    public void visitJavadocToken(DetailNode tagNode) {
+        // No need to filter token because overridden getDefaultJavadocTokens ensures that we only
+        // receive JAVADOC_TAG DetailNode.
+        if (!isAnotherTagBefore(tagNode)
+                && !isOnlyTagInWholeJavadoc(tagNode)
+                && insufficientConsecutiveNewlines(tagNode)) {
+            log(tagNode.getLineNumber(),
+                    MSG_JAVADOC_TAG_LINE_BEFORE,
+                    tagNode.getChildren()[0].getText());
+        }
+    }
+
+    private static boolean isAnotherTagBefore(DetailNode tagNode) {
```

Comment:
@romani Shouldn't this method be required to have a javadoc?

Reply:
It is strange that our Checkstyle config is not barking on this.
I am not a fan of javadoc on non public methods, but our codebase is crazy at on this point.

Do we have javadoc in other classes

Reply:
I started an issue as this is why I pointed it out, all methods should have a javadoc.